apps/web/src/services/asset-inventory/collector/CollectorPage.vue

```diff
@@ -134,27 +110,35 @@ const collectorApiQueryHelper = new ApiQueryHelper()
 
 /* Components */
 const initCollectorList = async () => {
+    state.initLoading = true;
+    try {
+        await collectorPageStore.getCollectorList();
+        if (collectorPageState.totalCount > 0) {
+            state.hasCollectorList = true;
+        }
+    } catch (e) {
+        await collectorPageStore.$reset();
+    } finally {
+        state.initLoading = false;
+    }
+};
+const refreshCollectorList = async () => {
     collectorApiQueryHelper.setFilters(collectorPageStore.allFilters);
     try {
         await collectorPageStore.getCollectorList(collectorApiQueryHelper.data);
-        if (Object.keys(state.items).length > 0) {
-            await collectorPageStore.setCollectorList(state.items);
-        }
     } catch (e) {
         ErrorHandler.handleError(e);
-        await collectorPageStore.$reset();
     }
 };
-const handleChangeToolbox = async (options) => {
+const handleSelectedProvider = (providerName: string) => {
+    collectorPageStore.setSelectedProvider(providerName);
+    refreshCollectorList();
+};
+const handleChangeToolbox = (options) => {
     setApiQueryWithToolboxOptions(collectorApiQueryHelper, options);
-    await initCollectorList();
+    refreshCollectorList();
 };
 
-/* Watcher */
-watch(() => collectorPageState.selectedProvider, async () => {
-    await initCollectorList();
-});
-
 /* Unmounted */
 onUnmounted(() => {
     collectorPageStore.$reset();
```

Comment:
```suggestion
    collectorPageStore.$dispose();
```
suggestion: I recommend using `$dispose()` because it completely removes the store. You can also confirm its complete removal in the dev tools.
(`$reset()` is included in`$dispose`)